hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -485,6 +488,10 @@ private static Object convertValueForAvroLogicalTypes(Schema fieldSchema, Object
         return decimalConversion.fromBytes((ByteBuffer) fieldValue, fieldSchema,
             LogicalTypes.decimal(dc.getPrecision(), dc.getScale()));
       }
+    } else if (fieldSchema.getLogicalType() == LogicalTypes.timestampMicros()) {
+      return Instant.EPOCH.plus(Long.parseLong(fieldValue.toString()), ChronoUnit.MICROS).atZone(ZoneId.systemDefault()).toLocalDateTime();
+    } else if (fieldSchema.getLogicalType() == LogicalTypes.timestampMillis()) {
```

Comment:
can you please point me to testcase which is covering this.

Reply:
Have added the test `testGetNestedFieldValForLogicalTypes` in `TestHoodieAvroUtils`

Reply:
thanks. looks like we don't have coverage for LogicalTypes.timestampMicros(). have left a comment below.

Reply:
Added

Reply:
Why we also change the value of type `LogicalTypes.timestampMillis` from long to `LocalDataTime` ? What is the gain ?

Reply:
We have just tried to maintain consistent data types for logical types. Previously, LogicalTypes.date was represented as LocalDate but LogicalTypes.timestampMillis was represented at long.